client/common/client_cliprdr_file.c

```diff
@@ -152,6 +152,55 @@ struct cliprdr_file_context
 	BYTE client_data_hash[WINPR_SHA256_DIGEST_LENGTH];
 };
 
+struct escape_pair
+{
+	const char* what;
+	const char* escape;
+};
+
+static const struct escape_pair url_escape_codes[] = {
+	{ " ", "\%20" }, { "<", "\%3C" }, { ">", "\%3E" }, { "#", "\%23" },  { "%", "\%25" },
+	{ "{", "\%7B" }, { "}", "\%7D" }, { "|", "\%7C" }, { "\\", "\%5C" }, { "^", "\%5E" },
+	{ "~", "\%7E" }, { "[", "\%5B" }, { "]", "\%5D" }, { "`", "\%60" },  { ";", "\%3B" },
+	{ "/", "\%2F" }, { "?", "\%3F" }, { ":", "\%3A" }, { "@", "\%40" },  { "=", "\%3D" },
+	{ "&", "\%26" }, { "$", "\%24" }
+};
+
+static char unescape(const char* what, size_t* px)
+{
+	for (size_t y = 0; y < ARRAYSIZE(url_escape_codes); y++)
+	{
+		const struct escape_pair* pair = &url_escape_codes[y];
+		if (strncmp(what, pair->escape, 3) == 0)
+		{
+			*px += 2;
+			return *pair->what;
+		}
+	}
+
+	return *what;
+}
+
+static char* url_unescape(const char* str, size_t len)
```

Comment:
Is it maybe worth it to put this function under winPR utils, so it can be reused?

Reply:
might be worth a look.